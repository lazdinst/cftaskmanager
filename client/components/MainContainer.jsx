import React from 'react';

// Redux Store and Actions
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getAllTasks } from '../store/modules/app';

// Import Local Components
import TaskList from './TaskList.jsx';
import TaskListHeader from './TaskListHeader.jsx';

// Import Semantic Components
import { Container, Divider } from 'semantic-ui-react';

export class MainContainer extends React.Component{
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Container text style={{ marginTop: '3em' }} >
        <TaskListHeader />
        <Divider />
        <TaskList />
      </Container>
    );
  }
}

const mapStateToProps = (state) => ({
  tasks: state.tasks,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  getAllTasks
}, dispatch);

export default connect( mapStateToProps, mapDispatchToProps)(MainContainer);