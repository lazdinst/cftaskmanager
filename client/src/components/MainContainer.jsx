import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { Container, Header } from 'semantic-ui-react';

import TaskList from './TaskList.jsx';
import TaskListHeader from './TaskListHeader.jsx';


import { getAllTasks } from '../store/modules/app';

// Define the main app class
class MainContainer extends React.Component{
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getAllTasks();
  }

  render() {
    return (
      <Container text style={{ marginTop: '3em' }}>
        <TaskListHeader />
        <TaskList />
      </Container>
    );
  }
}

const mapStateToProps = (state) => ({
  tasks: state.tasks,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  getAllTasks
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MainContainer);