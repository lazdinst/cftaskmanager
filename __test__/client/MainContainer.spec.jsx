import React from 'react'
import { shallow, mount } from 'enzyme';
import MainContainer from '../../client/components/MainContainer.jsx'
import toJson from 'enzyme-to-json';

describe('<MainContainer />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MainContainer />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});