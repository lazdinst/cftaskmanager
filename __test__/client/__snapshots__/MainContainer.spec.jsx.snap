// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> should render 1`] = `
<Container
  style={
    Object {
      "marginTop": "4em",
    }
  }
  text={true}
>
  <Connect(TaskListHeader) />
  <Divider />
  <Connect(TaskList) />
</Container>
`;
